types/plan/plan.go

```diff
@@ -177,15 +178,57 @@ func (output *KubernetesOutput) Merge(newoutput KubernetesOutput) {
 
 // Inputs defines the input section of plan
 type Inputs struct {
+	AbsRootDir          string                                   `yaml:"absRootDir"`
+	RelRootDir          string                                   `yaml:"relRootDir"`
```

Comment:
If these two variables are only for GetRelativePath function, at the expense of some extra operations in that function, we can potentially not have it in plan. Even if we REALLY need to have it in plan, I think these values need not be searialized into the plan.

Reply:
No, it's not about optimization, it's about correctness.
We are doing a lot of path manipulation in a large number of files. It's really hard to reason about the correctness
of those manipulations. We even fixed a few bugs related to this much earlier.
These 2 variables reduce the amount of manipulations that has be done as well make it easier to write code that will work in all situations.

Reply:
Plan to refactor a lot of things to start using these 2.

Reply:
We are already serializing the root directory, is there a reason to leave these 2 out?
```
RootDir             string                                   `yaml:"rootDir"`
```

Reply:
Since plan file is meant to be edited by users we should try to avoid havinf multiple fields which need to be consistent with each other. One possibly is to have a unserialized field absrootdir and use that always inside code and one serialized field that is relative folder (so that the file can be moved around along with source code without editing), which is used only for serialization. Always access absrootdir through a function call which checks if it is empty and fills it as required.

Reply:
@HarikrishnanBalagopal What is our plan for this?

Reply:
I have added `yaml:"-"` tags to both `AbsRootDir` and `RelRootDir` so they will be ignored during serialization.

Reply:
Also calling `plan.Spec.Inputs.SetRootDir` at the points of plan creation and also loading in `translate.go` to clean up the path and also set these 2 variables.